IO/BeagleBone/CANBusBBB.cs

```diff
@@ -135,14 +180,53 @@ internal CANBusBBB(string CanName)
 		/// <returns>A tuple, with the first element being the ID of the received CAN frame and the second being the payload</returns>
 		public Tuple<uint, byte[]> Read()
 		{
-			CANFrame Frame = new CANFrame();
-			read(Socket, ref Frame, Marshal.SizeOf(Frame));
-			byte[] Payload = new byte[Frame.DataLength];
-			unsafe
+			if (Extended)
 			{
-				for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+				ExtendedCANFrame Frame = new ExtendedCANFrame();
+				read(Socket, ref Frame, Marshal.SizeOf(Frame));
+				byte[] Payload = new byte[Frame.DataLength];
+				unsafe
+				{
+					for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+				}
+				return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+			}
+			else
+			{
+				CANFrame Frame = new CANFrame();
+				read(Socket, ref Frame, Marshal.SizeOf(Frame));
+				byte[] Payload = new byte[Frame.DataLength];
+				unsafe
+				{
+					for (int i = 0; i < Frame.DataLength; i++) { Payload[i] = Frame.Data[i]; }
+				}
+				return new Tuple<uint, byte[]>(Frame.CANID, Payload);
 			}
-			return new Tuple<uint, byte[]>(Frame.CANID, Payload);
+		}
+
+		private byte DLCToLen(byte CanDLC)
+		{
+			byte[] Dlc2Len = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 12, 16, 20, 24, 32, 48, 64 };
+			return Dlc2Len[CanDLC & 0x0F];
+		}
+
+		private byte LenToDLC(byte CanLen)
```

Comment:
Same here. Assuming this is a length, name it as such. Also provide a comment as to what this is doing. What is `DLC`?